src/platform_impl/macos/app_state.rs

```diff
@@ -248,6 +251,14 @@ impl AppState {
         }));
     }
 
+    pub fn set_file_open_callback<F>(callback: Option<F>)
+    where
+        F: FnMut(Vec<PathBuf>) -> FileOpenResult + Send + 'static,
+    {
+        *HANDLER.file_open_callback.lock().unwrap() =
+            callback.map(|c| Box::new(c) as Box<dyn FnMut(Vec<PathBuf>) -> FileOpenResult + Send>);
```

Comment:
```suggestion
            callback.map(|c| Box::new(c) as _);
```
I believe you can elide some or all of the type after `as` here, though I leave it at your discretion if that's better or worse for readability.

Reply:
Yes, this works!